docs/basic_training/cache_and_resume.md

```diff
@@ -249,63 +249,35 @@ Channel
+While dataflow channel ordering is guaranteed (i.e. data is read in the same order in which it’s written in the channel), there is no guarantee that the elements will be placed in the same respective position in the process output channel. This happens due to parallelism where multiple tasks may run at the same time, leading the operation on the second element to end sooner than the operation on the first element, for example.
 
 In practical terms, consider the following snippet:
 
 ```groovy linenums="1"
 process foo {
-    input:
-    tuple val(pair), path(reads)
+  input:
+    val x
+  output:
+    tuple val(task.index), val(x)
 
-    output:
-    tuple val(pair), path('*.bam'), emit: bam_ch
-
-    script:
+  script:
     """
-    your_command --here
+    sleep \$((RANDOM % 3))
     """
 }
 
-process bar {
-    input:
-    tuple val(pair), path(reads)
-
-    output:
-    tuple val(pair), path('*.bai'), emit: bai_ch
-
-    script:
-    """
-    other_command --here
-    """
-}
-
-process gather {
-    input:
-    tuple val(pair), path(bam)
-    tuple val(pair), path(bai)
-
-    script:
-    """
-    merge_command $bam $bai
-    """
+workflow {
+   channel.of('A','B','C','D') | foo | view
 }
 ```
 
-The inputs declared at line 29 and 30 can be delivered in any order because the execution order of the process `foo` and `bar` are not deterministic due to their parallel execution.
-
-Therefore the input of the third process needs to be synchronized using the [join](https://www.nextflow.io/docs/latest/operator.html#join) operator, or a similar approach. The third process should be written as:
-
-```groovy
-...
+Just like we saw at the beginning of this tutorial with HELLO WORLD or WORLD HELLO, the output of the snippet above can be:
 
-process gather {
-    input:
-    tuple val(pair), path(bam), path(bai)
-
-    script:
-    """
-    merge_command $bam $bai
-    """
-}
+```console
+[3, C]
+[4, D]
+[2, B]
+[1, A]
 ```
+
+Imagine now that we have output channels from two processes like this that are input channels to a third downstream process. If this third process expects the ordering to be the same, so that the first element in the first process output channel is related to the first element in the second process output channel, there will be a mismatch. An easy solution for this problem is to use the [fair](https://www.nextflow.io/docs/edge/process.html?highlight=fair#fair) process directive, which applies a fairness concept in multiprocessing that shares resources between the tasks in a fair way so that no task will get more computer resources than the other. Depending on your situation, using the `fair` directive will lead to a decrease in performance. Others have chosen to use tuples assigning an identification to channel elements (`['SRR123', '/some/path/SRR123.fasta']` instead of simply `/some/path/SRR123.fasta`) to make sure the processes are not incurring into a mismatch.
```

Comment:
```suggestion

..and that order will likely be different every time the pipeline is run.

Imagine now that we two processes like this, whose output channels are acting as input channels to a third process. Both channels will be independently random, so the third process must not expect them to retain a paired sequence. If it does assume that the first element in the first process output channel is related to the first element in the second process output channel, there will be a mismatch.

A common solution for this is to use what is commonly referred to as a _meta map_. A groovy object with sample information is passed out together with the file results within an output channel as a tuple. This can be then be used to pair samples from separate channels together for downstream use. For example, instead of putting just  `/some/path/myoutput.bam` into a channel, you could use `['SRR123', '/some/path/myoutput.bam']` to make sure the processes are not incurring into a mismatch.

If meta maps are not possible, an alternative is to use the [`fair`](https://nextflow.io/docs/edge/process.html#fair) process directive. When specified, Nextflow will guarantee that the order of outputs will match the order of inputs.

!!! warning

     Depending on your situation, using the `fair` directive will lead to a decrease in performance.
```

Reply:
Could be nice to give a better example of using meta maps here. I think that we should recommend `fair` as a last resort only. It's the lazy option, but it could have quite a negative impact if its use becomes widespread.

Reply:
Indeed! Can you point me to one? There are tons of examples in nf-core pipelines but if you have an easy and straightforward example at the tip of your tongue that would be great 😬

Reply:
Would that be a nice example?

```nextflow
// assume the channel below was generated with fromFilePairs
filepairs = [
    SRR493366,
    [/my/data/SRR493366_1.fastq, /my/data/SRR493366_2.fastq]
]
// A meta map would look like
meta_map = [
    [id: 'SRR493366', single_end: false], // meta map
    [/my/data/SRR493366_1.fastq, /my/data/SRR493366_2.fastq]
]
```